components/taxonomy/src/parse.rs

```diff
@@ -168,7 +178,15 @@ impl ParseError {
 }
 
 #[derive(Debug)]
-pub struct JSONError(error::Error);
+pub struct JSONError(pub error::Error);
+
+impl PartialEq for JSONError {
+    fn eq(&self, other: &JSONError) -> bool {
+        format!("{}", self.0) == format!("{}", other.0)
+    }
+}
+
+impl Eq for JSONError {}
 
 impl Clone for JSONError {
     fn clone(&self) -> JSONError {
```

Comment:
such a useful implementation ;) (not for this patch, as this is not changed here, but I was quite surprised by this so wanted to mention it)